docs/src/app/(public)/(content)/react/utils/use-render/page.mdx

```diff
@@ -0,0 +1,124 @@
+# useRender
+
+<Subtitle>Utility for adding Base UI-like features to custom components.</Subtitle>
+<Meta name="description" content="Utility for adding Base UI-like features to custom components." />
+
+The `useRender` hook allows you to support the same features Base UI provides across all components so that you can also have a consistent experience in your custom components:
+
+- A [render](/react/handbook/composition) prop to override the default rendered element (similar to `asChild` from Radix).
+- [data attributes](https://base-ui.com/react/handbook/styling#data-attributes) that map to the component's state which can be used for styling.
+- A [callback](/react/handbook/styling#css-classes) on the `className` prop that enables passing CSS classes depending on the component's state.
+
+## API reference
+
+### Input parameters
+
+<PropsReferenceTable
+  data={{
+    render: {
+      type: 'RenderProp<State>',
+      description:
+        "Allows you to replace the component's HTML element\nwith a different tag, or compose it with another component.\n\nAccepts a \`ReactElement\` or a function that returns the element to render.",
+    },
+    props: {
+      type: 'Record<string, unknown>',
+      description: 'Props to be spread on the rendered element.',
+    },
+    className: {
+      type: 'string | ((state: State) => string)',
+      description:
+        "CSS class applied to the element, or a function that returns a class based on the component's state.",
+    },
+    state: {
+      type: 'State',
+      description:
+        'The state of the component. It will be used as a parameter for the render and className callbacks.',
+    },
+    stateAttributesMap: {
+      type: 'StateAttributes<State>',
```

Comment:
AFAIK this type is not available for users so not sure if having `StateAttributes` here is helpful.